src/main/java/org/sonarsource/sonarlint/ls/AnalysisTaskExecutor.java

```diff
@@ -161,14 +162,23 @@ private void clearIssueCacheAndPublishEmptyDiagnostics(URI f) {
 
   private void analyze(AnalysisTask task, Optional<WorkspaceFolderWrapper> workspaceFolder, Map<URI, VersionedOpenFile> filesToAnalyze) {
     if (workspaceFolder.isPresent()) {
-      // We can only have the same binding for a given folder
+
+      var splitByNotebookOrNot = filesToAnalyze.values().stream()
```

Comment:
Maybe `fileGroupsByFileType` or similar is a more descriptive name? `splitByNotebookOrNot` is hard for me to grasp (I speak I 😛)